### ISSUE TITLE
Support for push?

### ISSUE THREAD

**Issue by tinoheth** [brettlangdon/git-vendor#6]
Hi there!
It's time for my annual study for git-supported code reuse ;-), and right now, I'm testing vendor.
I like its simplicity, but there's one feature that I am missing:
Git-lib (https://github.com/tomlu/git-lib) has support for push of "sub-repos", and this feature fits good into my workflow.
What's your opinion on adding a push-feature to vendor?

**Comment by brettlangdon** [comment]
hey @tinoheth what is your workflow? I am assuming to have multiple sub-projects live under one main repo, but being able to simultaneously work on, commit, and push to all sub-projects from the main project?

I have to read into `git-subtree push` a little more, and possibly play with it. I am not sure I fully understand how splitting works. My first impressions are that it probably won't work very well with `git-vendor`, since it sounds like it clones the history of the subtree. The problem with that is that `git-vendor` squashes the sub-repos history into a single commit when pulling it into our repo, so I am not sure how well `git-subtree split` will handle this, if it means we will be pushing the squashed commits to our sub-projects remote, then... we definitely wouldn't want that, since we'll get nothing but conflicts when pushing.

I am more than happy to dig into this a little further, but I will await your response concerning your workflow first, want to make sure I understand the problem.

**Comment by tinoheth** [comment]
I'm not sure how common this approach is, but for me, the perfect workflow would allow me to decide that a bunch of source files is useful for other projects as well, and make it easy to separate those files from the parent-project and evolve them in parallel (it's possible already, but quite tedious...).

When I change a library, this is most likely because I encountered a limitation in a project that uses this library, so it would be very convenient to have the ability to simply extend the sub-project, and push those changes afterwards.

Don't feel pushed to add those capabilities if you see no need for them - I just started playing with manually splitting subtrees and pushing those, and still have to see how comfortable the handling of commit-messages is...

**Comment by brettlangdon** [comment]
hmm, yeah. I mean, it seems like a perfect use case for `git-submodule`, since you'd be able to even create new branches, and easily run the equivalent `git pull` on the submodules. However, using `git-submodule` is sort of a beast of it's own, which I am personally not a fan.

**Comment by tinoheth** [comment]
That's right, git-submodule is a sad story: If this feature had better support...
One advantage of all subtree-solutions is that they "just work", even if a repo-user has no idea of any modules at all.

**Comment by brettlangdon** [comment]
Exactly.

I'll play around with `git-subtree push` a bit this week, see if I can learn more. If we can find a sensible workflow that works for `git-vendor`, I wouldn't be opposed to adding in the feature.

**Comment by nikolay** [comment]
[git-subrepo](https://github.com/ingydotnet/git-subrepo) already has push capability.
